diff --git a/jcl/source/common/JclFileUtils.pas b/jcl/source/common/JclFileUtils.pas
index cd4a966a2..031452d9e 100644
--- a/jcl/source/common/JclFileUtils.pas
+++ b/jcl/source/common/JclFileUtils.pas
@@ -4298,7 +4298,6 @@ var
   FileAttributesEx: WIN32_FILE_ATTRIBUTE_DATA;
   Size: TJclULargeInteger;
 begin
-  Result := 0;
   if GetFileAttributesEx(PChar(FileName), GetFileExInfoStandard, @FileAttributesEx) then
   begin
     Size.LowPart := FileAttributesEx.nFileSizeLow;
diff --git a/jcl/source/common/JclMath.pas b/jcl/source/common/JclMath.pas
index 8d1102916..a88ad2807 100644
--- a/jcl/source/common/JclMath.pas
+++ b/jcl/source/common/JclMath.pas
@@ -1360,7 +1360,6 @@ function ArcTan(X: Float): Float;
 begin
   {$IFDEF USE_MATH_UNIT}
   System.Error(rePlatformNotImplemented);
-  Result := NaN; 
   {$ELSE ~USE_MATH_UNIT}
   asm
           FLD     X
@@ -1405,7 +1404,6 @@ begin
   DomainCheck(Abs(X) > MaxAngle);
   {$IFDEF USE_MATH_UNIT}
   System.Error(rePlatformNotImplemented);
-  Result := NaN;
   {$ELSE ~USE_MATH_UNIT}
   asm
           FLD     X
@@ -1484,7 +1482,6 @@ begin
   {$ENDIF ~MATH_EXT_SPECIALVALUES}
   {$IFDEF USE_MATH_UNIT}
   System.Error(rePlatformNotImplemented);
-  Result := NaN;
   {$ELSE ~USE_MATH_UNIT}
   asm
           FLD     X
@@ -4742,7 +4739,6 @@ begin
   {$IFDEF DELPHI64_TEMPORARY}
   //IsInfinite is disabled for 64-bit, because BASM is not 64-bit compatible (see logmessage of public repo @3070)
   System.Error(rePlatformNotImplemented);
-  Result := False;
   {$ELSE ~DELPHI64_TEMPORARY}
   Result := JclMath.IsInfinite(Self);
   {$ENDIF ~DELPHI64_TEMPORARY}
@@ -4837,7 +4833,6 @@ begin
   {$IFDEF DELPHI64_TEMPORARY}
   //IsInfinite is disabled for 64-bit, because BASM is not 64-bit compatible (see logmessage of public repo @3070)
   System.Error(rePlatformNotImplemented);
-  Result := False;
   {$ELSE ~DELPHI64_TEMPORARY}
   Result := JclMath.IsInfinite(Self);
   {$ENDIF ~DELPHI64_TEMPORARY}
